### PR TITLE
FEATURE: implements generics filters for reports

### PR DESCRIPTION
**Pull request by jjaffeux**
category,group and file extension now use it.

### CODE REVIEW

**Comment by jjaffeux** [comment]
This PR allows custom filters to be added very simply:

Screenshot 2019-04-05 at 18 29 28

To make it possible you have to add support for in your report:

```
report.add_filter("filter-name")

# make use of you filter-name later in the report
filter_name = report.filter.dig("filter-name")

data = User.where(something: filter_name)
```

And creates a `report-filters/filter-name` component

It will give you out of the box: query params support (in the filters query param), UI support and caching. 

Meaning you can now link to : `/admin/reports/signups?filters=filter-name:xxx`

**Comment by eviltrout** [comment]
The PR is quite big, but one that raises my eyebrow is that the filters param seems to have a "micro format". These things can be somewhat of a trap. For example, what if the filter value need to contain a colon?

I can think of two potentially more robust solutions:

1. Use the Rails-style `filters[name]=value` format. This will pass automatically to rails as `"filters" => { "name" => "value" }` for easy parsing.

2. You could use `JSON.stringify` and put the parameters in a JSON object instead. That will making parsing/encoding it quite easy.

**Comment by jjaffeux** [comment]
@eviltrout fully agreed, I tried to implement 1) before going with this solution, but discovered it was not working out of the box in Ember and would have to go for a lot of parsing. Given my solution could be changed later if it was not working for a specific case I decided to not solve an issue I didn’t have.

But I didn’t think of 2) and it seems both well working and easy to do, so I implemented it in my last commit.

File: spec/models/report_spec.rb [hunk 1]
```diff
@@ -753,12 +753,12 @@
       end
 
       context "with category filtering" do
-        let(:report) { Report.find('flags', category_id: c1.id) }
+        let(:report) { Report.find('flags', filters: "category:#{c1.id}") }
```
Comment (eviltrout) [inline spec/models/report_spec.rb]: These aren't JSON?

File: test/javascripts/acceptance/dashboard-test.js.es6 [hunk 2]
```diff
@@ -86,3 +98,17 @@ QUnit.test("reports tab", async assert => {
     "filter is case insensitive"
   );
 });
+
+QUnit.test("report filters", async assert => {
+  await visit(
+    "/admin/reports/signups?end_date=2018-07-16&filters=group:88&start_date=2018-06-16"
```
Comment (eviltrout) [inline test/javascripts/acceptance/dashboard-test.js.es6]: Ditto here, I thought it was moved to JSON?

File: spec/models/report_spec.rb (hunk 1 above)
Comment (jjaffeux) [reply spec/models/report_spec.rb]: Was working on it, yes it's changed

File: test/javascripts/acceptance/dashboard-test.js.es6 (hunk 2 above)
Comment (jjaffeux) [reply test/javascripts/acceptance/dashboard-test.js.es6]: Working on it, I discovered an issue with caching, will ping you when ready for review.

**Pull request merged**